### PR TITLE
chapter_6

### PR DESCRIPTION
**Pull request by FirdausIsa**
ready

### CODE REVIEW

**Comment by FirdausIsa** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,16 @@
 class User < ApplicationRecord
+  before_save :downcase_email
+
+  private
+
+  def downcase_email
+  self.email.downcase!
+  end
+
+  validates :name, presence: true
```
Comment (ngocvt-0484) [inline app/models/user.rb:10]: please move validate and has_secure_password  before "private"

File: app/models/user.rb [hunk 2]
```diff
@@ -1,2 +1,16 @@
 class User < ApplicationRecord
+  before_save :downcase_email
+
+  private
+
+  def downcase_email
+  self.email.downcase!
```
Comment (ngocvt-0484) [inline app/models/user.rb:7]: wrong indent, please tab 2 space

**Comment by FirdausIsa** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged
You can merge this pull request

**Pull request merged**